app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/Matchers.ts

```diff
@@ -0,0 +1,170 @@
+import { Platform } from 'react-native';
+import { color, defaultTestErrorLog } from './logMessageUtils';
+import { ComparisonMode, TestCase, TrackerCallCount } from './types';
+
+type TestValue =
+  | TrackerCallCount
+  | string
+  | Array<unknown>
+  | number
+  | bigint
+  | Record<string, unknown>;
+
+export class Matchers {
+  constructor(private currentValue: TestValue, private testCase: TestCase) {}
+
+  private assertValueIsCallTracker(
+    value: any
+  ): asserts value is TrackerCallCount {
+    if (!('name' in value && 'JS' in value && 'UI' in value)) {
+      throw Error('Invalid value');
+    }
+  }
+
+  public toBe(
+    expectedValue: TestValue,
+    comparisonMode = ComparisonMode.DISTANCE
+  ) {
+    const CONDITIONS: { [Key: string]: (e: any, v: any) => Boolean } = {
+      [ComparisonMode.STRING]: (expected, value) => {
+        return value !== expected;
+      },
+      [ComparisonMode.NUMBER]: (e, v) => {
+        return isNaN(Number(v)) ? !isNaN(e) : Number(v) !== Number(e);
+      },
+      [ComparisonMode.COLOR]: (e, v) => {
+        const colorRegex = new RegExp('^#?([a-f0-9]{6}|[a-f0-9]{3})$');
+        if (!colorRegex.test(expectedValue as string)) {
+          throw Error(
+            `Invalid color format "${e}", please use lowercase hex color (like #123abc) `
+          );
+        }
+        return typeof v !== 'string' || v !== e;
+      },
+      [ComparisonMode.DISTANCE]: () => {
+        const valueAsNumber = Number(this.currentValue);
+        if (Platform.OS === 'ios') {
+          return (
+            isNaN(valueAsNumber) || valueAsNumber !== Number(expectedValue)
+          );
+        } else {
+          return (
+            isNaN(valueAsNumber) ||
+            Math.abs(valueAsNumber - Number(expectedValue)) > 1
```

Comment:
How about moving this `1` to some constant?

Reply:
We will move it to some function, like described above